lib/masamune/actions/invoke_parallel.rb

```diff
@@ -35,13 +35,16 @@ module InvokeParallel
     end
 
     def invoke_parallel(*task_group)
-      opts = task_group.last.is_a?(Hash) ? task_group.pop.dup : {}
-      max_tasks = [opts.delete(:max_tasks), task_group.count].min
+      per_task_opts = task_group.last.is_a?(Array) ? task_group.pop.dup : [{}]
```

Comment:
The use of `.pop.dup` isn't new here, but it seems like it wouldn't be necessary to create a copy of the object after popping it out of the array since the `pop` would remove the original object from the array in any case.  Does this protect against an unexpected condition of some sort?